File: jdk-1.5-parent/tinymce-parent/tinymce-examples/src/test/java/wicket/contrib/examples/tinymce/Start.java
package wicket.contrib.examples.tinymce;

import org.apache.wicket.util.time.Duration;
import org.eclipse.jetty.server.Server;
import org.eclipse.jetty.server.bio.SocketConnector;
import org.eclipse.jetty.webapp.WebAppContext;

public class Start
{
	public static void main(String[] args) throws Exception
	{
		int timeout = (int)Duration.ONE_HOUR.getMilliseconds();

		Server server = new Server();
		SocketConnector connector = new SocketConnector();

		// Set some timeout options to make debugging easier.
		connector.setMaxIdleTime(timeout);
		connector.setSoLingerTime(-1);
		connector.setPort(8081);
		server.addConnector(connector);

		// check if a keystore for a SSL certificate is available, and
		// if so, start a SSL connector on port 8443. By default, the
		// quickstart comes with a Apache Wicket Quickstart Certificate
		// that expires about half way september 2021. Do not use this
		// certificate anywhere important as the passwords are available
		// in the source.

		WebAppContext bb = new WebAppContext();
		bb.setServer(server);
		bb.setContextPath("/");
		bb.setWar("src/main/webapp");

		// START JMX SERVER
		// MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
		// MBeanContainer mBeanContainer = new MBeanContainer(mBeanServer);
		// server.getContainer().addEventListener(mBeanContainer);
		// mBeanContainer.start();

		server.setHandler(bb);

		try
		{
			System.out.println(">>> STARTING EMBEDDED JETTY SERVER, PRESS ANY KEY TO STOP");
			server.start();
			System.in.read();
			System.out.println(">>> STOPPING EMBEDDED JETTY SERVER");
			server.stop();
			server.join();
		}
		catch (Exception e)
		{
			e.printStackTrace();
			System.exit(1);
		}
	}
}

File: jdk-1.7-parent/scala-extensions-parent/wicket-scala-archetype/src/main/resources/archetype-resources/src/test/scala/Start.scala
package ${package}

import org.eclipse.jetty.server.Connector
import org.eclipse.jetty.server.Server
import org.eclipse.jetty.server.bio.SocketConnector
import org.eclipse.jetty.webapp.WebAppContext

object Start {

  def main(args:Array[String]): Unit = {
    val server = new Server
    val connector = new SocketConnector

    // Set some timeout options to make debugging easier.
    connector.setMaxIdleTime(1000 * 60 * 60)
    connector.setSoLingerTime(-1)
    connector.setPort(8080)
    server.setConnectors(Array[Connector](connector))

    val bb = new WebAppContext
    bb.setServer(server)
    bb.setContextPath("/")
    bb.setWar("src/main/webapp")

    // START JMX SERVER
    // val mBeanServer = ManagementFactory.getPlatformMBeanServer
    // val mBeanContainer = new MBeanContainer(mBeanServer)
    // server.getContainer().addEventListener(mBeanContainer)
    // mBeanContainer.start

    server.setHandler(bb)

    try {
      System.out.println(">>> STARTING EMBEDDED JETTY SERVER, PRESS ANY KEY TO STOP")
      server.start()
      System.in.read()
      System.out.println(">>> STOPPING EMBEDDED JETTY SERVER")
      server.stop()
      server.join()
    } catch {
      case x: Exception =>
        x.printStackTrace()
        System.exit(100)
    }
  }
}
